spec/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/cloud_database.spec

```diff
@@ -18,6 +18,20 @@ describe ManageIQ::Providers::IbmCloud::VPC::CloudManager::CloudDatabase do
 
     before { allow(ems).to receive(:with_provider_connection).and_yield(connection) }
 
+    context '#create_cloud_database' do
+      it 'creates the cloud database' do
+        allow(connection).to receive_message_chain(:resource, :manager, :collection, :find).and_return({:id => "test123"})
+        allow(connection).to receive_message_chain(:resource, :controller).and_return(resource_controller)
+        expect(resource_controller).to receive(:request).with(:create_resource_instance,
+                                                              :name             => "test-db",
+                                                              :target           => ems.provider_region,
+                                                              :resource_group   => "test123",
+                                                              :resource_plan_id => "databases-for-postgresql-standard")
+        cloud_database.class.raw_create_cloud_database(ems, {:name     => "test-db",
```

Comment:
Using `raw_create_cloud_database` here due to `missing credentials` error when using `create_cloud_database`

Reply:
Oh, that is because you are adding method mocks to the `ems` object here, but when you call `create_cloud_database` you pass an ID and it looks up the EMS thus returning a new object.  This new object doesn't have these mocks registered.

Sometimes we do `allow(ExtManagementSystem).to receive(:find).with(ems.id).and_return(ems)` but that's almost as gross IMO

This is just fine as a workaround